card/mod/01_core/set/all/templating.rb

```diff
@@ -12,29 +12,34 @@ def template
   # note that a *default template is never returned for an existing card.
   @template ||= begin
     @virtual = false
+
+    # NEW CARDS
     if new_card?
-      default_card = rule_card :default, :skip_modules=>true
+      default = rule_card :default, :skip_modules=>true
 
       dup_card = dup
-      dup_card.type_id = default_card ? default_card.type_id : Card.default_type_id
-
+      dup_card.type_id = default ? default.type_id : Card.default_type_id
 
-      if content_card = dup_card.structure_rule_card
+      if structure = dup_card.structure_rule_card
         @virtual = true if junction?
-        content_card
+        self.type_id = structure.type_id if assign_type_to?(structure)
```

Comment:
By definition, virtual cards are never real; this isn't a problem for real cards.  (Virtual cards are a subset of structured cards.)

It's actually not so much about caching/memory as initialization.  We weren't using the type from the structured card until the validation process, but since virtual cards are never saved (and thus never validated), it wasn't getting set correctly when there was an explicit type given.